cpp/models/abm/location_type.h

```diff
@@ -76,6 +77,48 @@ struct LocationId {
         }
         return (type < rhs.type);
     }
+
+    // same interface as location
+    uint32_t get_index() const
+    {
+        return index;
+    }
+
+    // same interface as location
+    LocationType get_type() const
+    {
+        return type;
+    }
```

Comment:
Do you need these two functions? Both members of LocationId are publicly accessible, right?

Reply:
No, they are not necessary. They do allow using the same interface for Location and its ID to access type/index, which saved me some additional work from changing several interfaces from using Location to LocationId. I could argue that getter/setters can be better than direct access, but these functions are here mostly for convenience.